nexus/src/db/datastore.rs

```diff
@@ -1028,6 +1034,262 @@ impl DataStore {
             })
     }
 
+    // IP Pools
+
+    /// List IP Pools by their name
+    pub async fn ip_pools_list_by_name(
+        &self,
+        opctx: &OpContext,
+        pagparams: &DataPageParams<'_, Name>,
+    ) -> ListResultVec<IpPool> {
+        use db::schema::ip_pool::dsl;
+        opctx
+            .authorize(authz::Action::ListChildren, &authz::IP_POOL_LIST)
+            .await?;
+        paginated(dsl::ip_pool, dsl::name, pagparams)
+            .filter(dsl::time_deleted.is_null())
+            .select(db::model::IpPool::as_select())
+            .get_results_async(self.pool_authorized(opctx).await?)
+            .await
+            .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
+    }
+
+    /// List IP Pools by their IDs
+    pub async fn ip_pools_list_by_id(
+        &self,
+        opctx: &OpContext,
+        pagparams: &DataPageParams<'_, Uuid>,
+    ) -> ListResultVec<IpPool> {
+        use db::schema::ip_pool::dsl;
+        opctx
+            .authorize(authz::Action::ListChildren, &authz::IP_POOL_LIST)
+            .await?;
+        paginated(dsl::ip_pool, dsl::id, pagparams)
+            .filter(dsl::time_deleted.is_null())
+            .select(db::model::IpPool::as_select())
+            .get_results_async(self.pool_authorized(opctx).await?)
+            .await
+            .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
+    }
+
+    pub async fn ip_pool_create(
+        &self,
+        opctx: &OpContext,
+        new_pool: &params::IpPoolCreate,
+    ) -> CreateResult<IpPool> {
+        use db::schema::ip_pool::dsl;
+        opctx
+            .authorize(authz::Action::CreateChild, &authz::IP_POOL_LIST)
+            .await?;
+        let pool = IpPool::new(&new_pool.identity);
+        let pool_name = pool.name().as_str().to_string();
+        diesel::insert_into(dsl::ip_pool)
+            .values(pool)
+            .on_conflict(dsl::id)
```

Comment:
Why this?  I think we usually do this in saga actions because they have to be idempotent with the same id, but here it seems like it'd be very surprising to get a conflict here.

Reply:
I think that was my fingers on a bit too much autopilot. I'll remove.